### PR TITLE
Sudoku!

### PR DESCRIPTION
**Pull request by NanakoAC**
Adds a sudoku game for modular computers, works on all platforms.

For those who've never heard of it, its a little game about logic and numbers. Simple but timeless, and helps keep the brain sharp in situations that would otherwise make you fall asleep.

Comes with a fast backtracking algorithm to generate unique puzzles, hints, and lots of general polish. Includes unique screen icons  for all three platforms

### CODE REVIEW

**Comment by LordFowl** [comment]
'france' and 'japan' should be capitalized into 'France' and 'Japan'

**Comment by Lohikar** [comment]
Could you maybe add a way to hide the description on the right? I doubt you'd really need to read it every time you want to play a game.

**Comment by NanakoAC** [comment]
fixed the capitalisation thing.

The description is mostly just there to fill space, i made the board use the same aspect ratio as a console screen so it roughly matches the screen image. if that description were hidden it'd just be a block of empty space instead

**Comment by Lohikar** [comment]
What about centering the game board when the description is hidden?
Should be doable with light CSS.

**Comment by NanakoAC** [comment]
eh okay, working on collapse

**Comment by NanakoAC** [comment]
Added an expand/collapse button.  When collapsed its like this

File: nano/templates/sudoku.tmpl [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+				<div style="font-size: 18px;">Start a New Game</div>
+				<hr>
+				<form name="newgame" method="get" action='?src={{:data.src}}'>
+					<input type='hidden' name='src' value='{{:data.src}}'>
+					<input type="radio" name="difficulty" value="Easy" checked> Easy<br>
+					<input type="radio" name="difficulty" value="Medium"> Medium<br>
+					<input type="radio" name="difficulty" value="Hard"> Hard<br>
+					<input type="radio" name="difficulty" value="Robust"> Robust<br>
+					<input type='submit' class="nanoformbutton" value='Generate New Grid'>
+				</form>
```
Comment (Arrow768) [inline nano/templates/sudoku.tmpl]: As discussed, this is a hack.
Use proper nanoui. Take the laptop vendor as an example for multiple options.
What you have done here has the following side effect:
If you walk away from the console a tile, you cant use any buttons except the Generate new grid. (which doesnt make sense)

Comment (NanakoAC) [inline nano/templates/sudoku.tmpl]: it is done

File: code/modules/modular_computers/file_system/programs/games/sudoku/sudoku.dm [hunk 2]
```diff
@@ -91,11 +94,20 @@
...
+			advanced_populate_grid(clues[newdifficulty])
+		else
+			return
+	/*else if (href_list["difficulty"])
 		var/response = alert(usr,"Are you sure you want to start a new game? All progress on this one will be lost. Be sure to pick your desired difficulty first.","New Puzzle","Start Anew","Wait no!")
 		if (response == "Start Anew")
 			advanced_populate_grid(clues[href_list["difficulty"]])
 		else
 			return
+	*/
```
Comment (Arrow768) [inline code/modules/modular_computers/file_system/programs/games/sudoku/sudoku.dm]: Why commented out and not removed ?

**Pull request merged**